### PR TITLE
CXP-1402: mapping - accept date as source for string-datetime target

### PR DESCRIPTION
**Pull request by hogren**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by guillaumefer-akeneo — APPROVED** [review]

**Review by masacc — COMMENTED** [review]
You have to be absolutely sure that what you get from `raw_values` is a date in ISO8601 format in `StringDateTimeFromDateAttributeValueExtractor`. 

For now the only check that is done is that the value is a string :
```
return \is_string($value) ? $value : null;
```

File: components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AttributeSource/AttributeDateSource.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                            new Assert\NotBlank(),
+                        ],
+                        'scope' => [
+                            new Assert\AtLeastOneOf([
+                                new Assert\IsNull(),
+                                new Assert\Sequentially([
+                                    new Assert\Type('string'),
+                                    new Assert\NotBlank(),
+                                ]),
+                            ]),
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AttributeSource/AttributeDateSource.php:43]:
```suggestion
                            new Assert\Type('string'),
                            new Assert\NotBlank(allowNull: true),
```

Comment (hogren) [reply components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AttributeSource/AttributeDateSource.php]: Thanks for the tip ! @masacc

File: components/catalogs/back/src/Application/Service/TargetTypeConverter.php [hunk 2]
```diff
@@ -18,6 +18,9 @@ final class TargetTypeConverter
             'pim_catalog_textarea',
```
Comment (masacc) [inline components/catalogs/back/src/Application/Service/TargetTypeConverter.php:18]: I moved this file in `Application/Mapping/TargetTypeConverter.php` (Sercice -> Mapping). If you have any conflict, be aware of that. Same for its associated TU.

File: components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php [hunk 3]
```diff
@@ -47,6 +47,13 @@ public function validConversionProvider(): array
                     'pim_catalog_simpleselect',
                 ],
             ],
+            'date' => [
```
Comment (masacc) [inline components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php:50]:
```suggestion
            'string+date-time' => [
```

**Review by masacc — APPROVED** [review]

**Review by masacc — APPROVED** [review]

**Pull request merged**